elm/wizard.py

```diff
@@ -311,3 +238,310 @@ def chat(self, query,
+        db_password = os.getenv('EWIZ_DB_PASSWORD')
+        assert db_user is not None, "Must set user for postgres database!"
+        assert db_password is not None, "Must set user for postgres database!"
+
+        self.conn = psycopg2.connect(user=db_user,
+                                     password=db_password,
+                                     host=db_host,
+                                     port=db_port,
+                                     database=db_name)
+
+        self.cursor = self.conn.cursor()
+
+        access_key = os.getenv('AWS_ACCESS_KEY_ID')
+        secret_key = os.getenv('AWS_SECRET_ACCESS_KEY')
+        session_token = os.getenv('AWS_SESSION_TOKEN')
+
+        assert access_key is not None, "Must set AWS_ACCESS_KEY_ID!"
+        assert secret_key is not None, ("Must set AWS_SECRET_ACCESS_KEY!")
+        assert session_token is not None, "Must set AWS_SESSION_TOKEN!"
+        self._aws_client = boto3.client(service_name='bedrock-runtime',
+                                        region_name='us-west-2',
+                                        aws_access_key_id=access_key,
+                                        aws_secret_access_key=secret_key,
+                                        aws_session_token=session_token)
+
+        super().__init__(model, token_budget=token_budget)
+
+    def get_embedding(self, text):
+        """Get the 1D array (list) embedding of a text string
+        as generated by AWS Titan.
+
+        Parameters
+        ----------
+        text : str
+            Text to embed
+
+        Returns
+        -------
+        embedding : list
+            List of float that represents the numerical embedding of the text
+        """
+
+        body = json.dumps({"inputText": text, })
+
+        model_id = self.EMBEDDING_MODEL
+        accept = 'application/json'
+        content_type = 'application/json'
+
+        response = self._aws_client.invoke_model(
+            body=body,
+            modelId=model_id,
+            accept=accept,
+            contentType=content_type
+        )
+
+        response_body = json.loads(response['body'].read())
+        embedding = response_body.get('embedding')
+
+        return embedding
+
+    def query_vector_db(self, query, limit=100):
+        """Returns a list of strings and relatednesses, sorted from most
+        related to least.
+
+        Parameters
+        ----------
+        query : str
+            Question being asked of GPT
+        limit : int
+            Number of top results to return.
+
+        Returns
+        -------
+        strings : np.ndarray
+            1D array of related strings
+        score : np.ndarray
+            1D array of float scores of strings
+        ids : np.ndarray
+            1D array of IDs in the text corpus corresponding to the
+            ranked strings/scores outputs.
+        """
+
+        query_embedding = self.get_embedding(query)
+
+        self.cursor.execute("SELECT ewiz_kb.id, "
```

Comment:
What is `ewiz_kb`? Is that a table name? The user should be able to modify this as a class attribute or an init parameter. Same with `ewiz_schema`